specification/metrics/semantic_conventions/runtime-environment-metrics.md

```diff
@@ -398,5 +402,83 @@ This metric is obtained from [`BufferPoolMXBean#getCount()`](https://docs.oracle
 **[1]:** Pool names are generally obtained via [BufferPoolMXBean#getName()](https://docs.oracle.com/en/java/javase/11/docs/api/java.management/java/lang/management/BufferPoolMXBean.html#getName()).
 <!-- endsemconv -->
 
+### Metric: `process.runtime.jvm.cpu.monitor.duration`
+
+This metric is [recommended][MetricRecommended]. Only available with JDK 17+.
+This metric is obtained from [`jdk.JavaMonitorWait`](https://sap.github.io/SapMachine/jfrevents/21.html#javamonitorwait) and [`jdk.JavaMonitorEnter`](https://sap.github.io/SapMachine/jfrevents/21.html#javamonitorenter) JFR events.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advice)
+of `[]` (single bucket histogram capturing count, sum, min, max).
+
+<!-- semconv metric.process.runtime.jvm.cpu.monitor.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `process.runtime.jvm.cpu.monitor.duration` | Histogram | `s` | Time monitor was used by a thread. Only available in JDK 17+. |
+<!-- endsemconv -->
+
+<!-- semconv metric.process.runtime.jvm.cpu.monitor.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `class` | string | Class of the monitor. | `java.lang.Object` | Opt-In |
+| `state` | string | Action taken at monitor. | `blocked`; `wait` | Recommended |
+| [`thread.id`](../../trace/semantic_conventions/span-general.md) | int | Current "managed" thread ID (as opposed to OS thread ID). | `42` | Opt-In |
+<!-- endsemconv -->
+
+### Metric: `process.runtime.jvm.cpu.context_swtich`
+
+This metric is [recommended][MetricRecommended]. Only available with JDK 17+.
+This metric is obtained from [`jdk.ThreadContextSwitchRate`](https://sap.github.io/SapMachine/jfrevents/21.html#threadcontextswitchrate) JFR events.
+
+<!-- semconv metric.process.runtime.jvm.cpu.context_switch(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `process.runtime.jvm.context_switches` | UpDownCounter | `Hz` | Number of context switches per second. Only available in JDK 17+. |
+<!-- endsemconv -->
+
+### Metric: `process.runtime.jvm.network.io`
+
+This metric is [recommended][MetricRecommended]. Only available with JDK 17+.
+This metric is obtained from [`jdk.SocketWrite`](https://sap.github.io/SapMachine/jfrevents/21.html#socketwrite) and [`jdk.SocketRead`](https://sap.github.io/SapMachine/jfrevents/21.html#socketread) JFR events.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advice)
+of `[]` (single bucket histogram capturing count, sum, min, max).
```

Comment:
Couldn't we define some buckets to understand better if our bytes are falling within standard packet sizes?

I'm not sure I understand this metric so well.  What would `count` be here?  Is `count` the number of packets *or* the number of times JFR reports data (i.e. relativley useless to a user).

If the later, why isn't this just a Counter of some sort?

Reply:
I'd encourage thinking about setting baseline buckets to common network packet sizes here.  I think the value of even 1-2 buckets would dramatically increase the overall value of this metric.

Hell, even understanding if the bytes written tend to be "larger than fits on a packet" or "smaller than fits on a packet" is a useful count to track.